tests/test_integration_api2_perturbed_maximization.py

```diff
@@ -0,0 +1,1142 @@
+"""
+
+import json
+
+import openreview
+from openreview.api import Note
+import pytest
+
+from conftest import clean_start_conference_v2, wait_for_status, post_perturbed_to_api2
+
+@pytest.fixture(scope="module")
+def venue(openreview_context):
+        openreview_client = openreview_context["openreview_client_v2"]
+
+        conference_id = "PTBMX.ws/2024/Conference"
+        num_reviewers = 10
+        num_papers = 10
+
+        venue = clean_start_conference_v2(
+            openreview_client,
+            conference_id,
+            num_reviewers,
+            num_papers,
+            0
+        )
+        post_perturbed_to_api2(openreview_client, conference_id)
+
+        assert len(openreview_client.get_group('PTBMX.ws/2024/Conference/Reviewers').members) == num_reviewers
+        assert len(openreview_client.get_notes(invitation='PTBMX.ws/2024/Conference/-/Submission')) == num_papers
+        return venue
+
+def test_integration_basic(openreview_context, venue, celery_app, celery_session_worker):
+    """
+    Basic integration test. Makes use of the OpenReview Builder
+    """
+    openreview_client = openreview_context["openreview_client_v2"]
+    test_client = openreview_context["test_client"]
+
+    num_papers = 10    
+    reviews_per_paper = 3
+    max_papers = 5
+    min_papers = 1
+    alternates = 0
+
+
+    reviewers_id = venue.get_reviewers_id()
+
+    config = {
+        "title": {"value": "integration-test"},
+        "user_demand": {"value": str(reviews_per_paper)},
+        "max_papers": {"value": str(max_papers)},
+        "min_papers": {"value": str(min_papers)},
+        "alternates": {"value": str(alternates)},
+        "config_invitation": {
+            "value": "{}/-/Assignment_Configuration".format(reviewers_id)
+        },
+        "paper_invitation": {"value": venue.get_submission_id()},
+        "assignment_invitation": {
+            "value": venue.get_assignment_id(reviewers_id)
+        },
+        "deployed_assignment_invitation": {
+            "value": venue.get_assignment_id(reviewers_id, deployed=True)
+        },
+        "invite_assignment_invitation": {
+            "value": venue.get_assignment_id(reviewers_id, invite=True)
+        },
+        "aggregate_score_invitation": {
+            "value": "{}/-/Aggregate_Score".format(reviewers_id)
+        },
+        "conflicts_invitation": {
+            "value": venue.get_conflict_score_id(reviewers_id)
+        },
+        "custom_max_papers_invitation": {
+            "value": "{}/-/Custom_Max_Papers".format(reviewers_id)
+        },
+        "match_group": {"value": reviewers_id},
+        "scores_specification": {
+            "value": {
+                venue.get_affinity_score_id(reviewers_id): {
+                    "weight": 1.0,
+                    "default": 0.0,
+                }
+            }
+        },
+        "status": {"value": "Initialized"},
+        "solver": {"value": "PerturbedMaximization"},
```

Comment:
@haroldrubio this value doesn't seem to match with the class name, I don't think this test is using this solver, it is using the MinMax as default.

Reply:
Though not completely sure, I think the usage seems to be correct. 

There is a SOLVER_MAP in matcher/core.py to convert this string to the actual solver class name. 

```
SOLVER_MAP = {
    "MinMax": MinMaxSolver,
    "FairFlow": FairFlow,
    "Randomized": RandomizedSolver,
    "FairSequence": FairSequence,
    "FairIR": FairIR,
    "PerturbedMaximization": PerturbedMaximizationSolver
}
```

Also, in tests/test_integration_api2_randomized.py, I can also see similar usage, while the actual solver class name is RandomizedSolver.

```
        "status": {"value": "Initialized"},
        "solver": {"value": "Randomized"},
        "randomized_probability_limits": {"value": "0.5"},
    }
```

Reply:
You are right, it is weird the API is not failing because `PerturbedMaximization` is not added to the Assignment_Configuration invitation as an available solver. I will check with Harold.